presto-main/src/main/java/com/facebook/presto/spiller/PartitioningSpillerFactory.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spiller;
+
+import com.facebook.presto.memory.AggregatedMemoryContext;
+import com.facebook.presto.operator.SpillContext;
+import com.facebook.presto.operator.exchange.LocalPartitionGenerator;
+import com.facebook.presto.spi.type.Type;
+
+import java.util.List;
+import java.util.function.Supplier;
+
+public interface PartitioningSpillerFactory
+{
+    PartitioningSpiller create(
+            List<Type> types,
+            LocalPartitionGenerator partitionGenerator,
+            int partitionsCount,
+            Supplier<SpillContext> spillContextSupplier,
```

Comment:
It should just take `SpillContext` which is a factory for child `SpillContext` instances itself.